gadopt/inverse.py

```diff
@@ -359,3 +360,54 @@ def __getstate__(self):
         "Iteration Limit": 100,
     },
 }
+
+
+class DiagnosticBlock(Block):
+    def __init__(self, f, function, riesz_options={'riesz_representation': 'L2'}):
+        super().__init__()
+        self.add_dependency(function)
+        self.add_output(function.block_variable)
+        self.f = f
+        self.f_name = function.name()
+        self.riesz_options = riesz_options
+
+    def recompute_component(self, inputs, block_variable, idx, prepared):
+        return block_variable.checkpoint
+
+    def evaluate_adj_component(self, inputs, adj_inputs, block_variable, idx, prepared=None):
+        out = inputs[0]._ad_convert_type(adj_inputs[0], options=self.riesz_options)
+        out.rename('adjoint_'+self.f_name)
+        self.f.write(out)
+        return 0
+
+
+class DiagnosticConstantBlock(Block):
```

Comment:
Docstring: What/why is this block?

Reply:
deleted this as dont need it for this PR. I think it would be neater to combine function/constant together to save redundant code anyway so i guess we can fix this when we have a need for it!